packages/tools/src/eventListeners/mouse/mouseDownListener.ts

```diff
@@ -176,52 +237,186 @@ function _onMouseDrag(evt: MouseEvent) {
+    clearTimeout(state.preventClickTimeout);
+
+    const eventName = state.isClickEvent ? MOUSE_CLICK : MOUSE_UP;
+
+    const currentPoints = getMouseEventPoints(evt, state.element);
+    const deltaPoints = _getDeltaPoints(currentPoints, state.lastPoints);
+    const eventDetail:
+      | EventTypes.MouseUpEventDetail
+      | EventTypes.MouseClickEventType = {
+      event: evt,
+      eventName,
+      mouseButton: state.mouseButton,
+      element: state.element,
+      renderingEngineId: state.renderingEngineId,
+      viewportId: state.viewportId,
+      camera: {},
+      startPoints: _copyPoints(state.startPoints),
+      lastPoints: _copyPoints(state.lastPoints),
+      currentPoints,
+      deltaPoints,
+    };
+
+    triggerEvent(eventDetail.element, eventName, eventDetail);
+
+    document.removeEventListener('mouseup', _onMouseUp);
 
-  triggerEvent(eventDetail.element, eventName, eventDetail);
+    state.element.removeEventListener('mousemove', _onMouseMove);
 
-  // Remove our temporary handlers
+    // Restore our global mousemove listener
+    state.element.addEventListener('mousemove', mouseMoveListener);
+
+    // Restore `state` to `defaultState`
+    state = JSON.parse(JSON.stringify(defaultState));
+  }
+
+  // Remove the drag as soon as we get the mouse up because either we have executed
+  // the mouse up logic, or we have not even handled the mouse down logic yet
+  // - either way no drag should/can occur.
   document.removeEventListener('mousemove', _onMouseDrag);
-  document.removeEventListener('mouseup', _onMouseUp);
+}
+
+/**
+ * Handles a mouse move on the state element after a mouse down AND up AND
+ * while the double click timeout is still running.
+ * @private
+ * @param evt - The mouse event.
+ */
+function _onMouseMove(evt: MouseEvent) {
+  const currentPoints = getMouseEventPoints(evt, state.element);
+  const lastPoints = _updateMouseEventsLastPoints(
+    state.element,
+    state.lastPoints
+  );
+
+  const deltaPoints = _getDeltaPoints(currentPoints, lastPoints);
+
+  if (!_isDragPastDoubleClickTolerance(deltaPoints.canvas)) {
+    return;
+  }
 
-  // Restore our global mousemove listener
-  state.element.addEventListener('mousemove', mouseMoveListener);
+  _doStateMouseDownAndUp();
 
-  // Restore `state` to `defaultState`
-  state = JSON.parse(JSON.stringify(defaultState));
+  // Moving past the tolerance means no double click should occur.
+  doubleClickState.ignoreDoubleClick = true;
+
+  // Do the move again because during the timeout the global mouse move listener was removed.
+  // Now it is back.
+  mouseMoveListener(evt);
+}
+
+/**
+ * Determines if the given delta is past the double click, drag distance tolerance.
+ * @param delta the delta
+ * @returns true iff the delta is past the tolerance
+ */
+function _isDragPastDoubleClickTolerance(delta: Types.Point2): boolean {
+  return (
+    Math.sqrt(Math.pow(delta[0], 2) + Math.pow(delta[1], 2)) >
```

Comment:
This is too much to get calculated for each mouse move, a lot of computations. Can we make it simpler to not rely on sqrt and pow?

Reply:
I understand and yes I can change it. However note that this is only for every move during the timeout. The timeout is relatively short. Also, once a drag/move goes past the tolerance the timeout is cleared. I would argue that yes it is expensive but for a very short time - likely less than the timeout if the mouse is moving any significant amount. Would you still like me to change it?